mp/src/game/client/hud_damageindicator.cpp

```diff
@@ -0,0 +1,332 @@
+static ConVar radius("mom_hud_damageindicator_radius", "120", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                     "How far away the damage indicators are from the crosshair.\n");
+static ConVar lifetime("mom_hud_damageindicator_lifetime", "1", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                       "How long the indicator stays on screen.\n");
+static ConVar dmgcolor("mom_hud_damageindicator_color", "980000", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                       "Color of the damage indicator.\n");
+
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+class CHudDamageIndicator : public CHudElement, public vgui::Panel
+{
+    DECLARE_CLASS_SIMPLE(CHudDamageIndicator, vgui::Panel);
+
+  public:
+    CHudDamageIndicator(const char *pElementName);
+    void Init(void);
+    void VidInit(void);
+    void Reset(void);
+    virtual bool ShouldDraw(void);
+
+    // Handler for our message
+    void MsgFunc_DamageIndicator(bf_read &msg);
+
+  private:
+    virtual void OnThink();
+    virtual void Paint();
+    virtual void ApplySchemeSettings(vgui::IScheme *pScheme);
+
+    // Painting
+    void GetDamagePosition(const Vector &vecDelta, float flRadius, float *xpos, float *ypos, float *flRotation);
+    void DrawDamageIndicator(int x0, int y0, int x1, int y1, float alpha, float flRotation);
+
+  private:
+    // Indication times
+    CPanelAnimationVar(float, m_iMaximumDamage, "MaximumDamage", "100");
+    CPanelAnimationVar(float, m_flFadeOutPercentage, "FadeOutPercentage", "0.7");
+
+    float m_flMinimumWidth;
+    float m_flMaximumWidth;
+    float m_flMinimumHeight;
+    float m_flMaximumHeight;
+    float m_flRadius;
+    float m_flMinimumTime;
+    float m_flMaximumTime;
+
+    // List of damages we've taken
+    struct damage_t
+    {
+        int iScale;
+        float flLifeTime;
+        float flStartTime;
+        Vector vecDelta; // Damage origin relative to the player
+    };
+    CUtlVector<damage_t> m_vecDamages;
+
+    CMaterialReference m_WhiteAdditiveMaterial;
+};
+
+DECLARE_HUDELEMENT(CHudDamageIndicator);
+DECLARE_HUD_MESSAGE(CHudDamageIndicator, DamageIndicator);
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+CHudDamageIndicator::CHudDamageIndicator(const char *pElementName)
+    : CHudElement(pElementName), BaseClass(NULL, "HudDamageIndicator")
+{
+    vgui::Panel *pParent = g_pClientMode->GetViewport();
+    SetParent(pParent);
+
+    m_WhiteAdditiveMaterial.Init("vgui/damage", TEXTURE_GROUP_VGUI);
+}
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
```

Comment:
You can remove these empty comment blocks above everything. Function/class names give the purpose of them anyways.